File: 김하영/Barkingdog_TwoPointer/2230.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [N, M] = input
  .shift()
  .split(" ")
  .map((str) => Number(str.trim()));
const inputArr = input.map((str) => Number(str.trim())).sort((a, b) => a - b);

const solution = (inputArr) => {
  let first = 0;
  let last = 1;

  let min = Infinity;

  while (first < inputArr.length && last < inputArr.length) {
    const diff = inputArr[last] - inputArr[first];

    if (diff < M) {
      last++;
    } else {
      min = Math.min(min, diff);
      first++;
    }
  }

  return min;
};

console.log(solution(inputArr));

File: BoSeonPark/BarkingDog_투포인터/2230.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

nums = [int(input()) for _ in range(n)]
nums.sort()
answer = 2000000001
s, e = 0, 0


while s < n and e < n:
    r = nums[e] - nums[s]
    
    if r >= m:
        s += 1
        answer = min(answer, r)
    else:
        e += 1

print(answer)